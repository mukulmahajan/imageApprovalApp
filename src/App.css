.App {
  text-align: center;
  background-color: #3b53e3;
  height: 100vh;
  width: 100%;
}
