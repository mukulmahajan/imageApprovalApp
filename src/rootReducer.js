import { combineReducers } from "redux";
import reducer from "./components/reducer/reducer";

const rootReducer = combineReducers({
  imagesDetails: reducer,
});

export default rootReducer;
