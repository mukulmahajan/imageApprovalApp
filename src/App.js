import "./App.css";
import { Nav } from "./components/Nav";
import { ApprovedImages } from "./components/approvedImages";
import { AddImage } from "./components/AddImage";
import { Footer } from "./components/Footer";
import { AppContainer } from "./components/styledComponents";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { setPreviousState } from "./components/actions/actions";

function App() {
  const initialState = useSelector((state) => state.imagesDetails);
  const dispatch = useDispatch();
  useEffect(() => {
    const serializedState = sessionStorage.getItem("sellics_images_state");
    if (!serializedState) {
      sessionStorage.setItem(
        "sellics_images_state",
        JSON.stringify(initialState)
      );
    } else {
      dispatch(setPreviousState(JSON.parse(serializedState)));
    }
  }, []);
  return (
    <div className="App">
      <AppContainer>
        <Nav />
        <ApprovedImages />
        <AddImage />
        <Footer />
      </AppContainer>
    </div>
  );
}

export default App;
