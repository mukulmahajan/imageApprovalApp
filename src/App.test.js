import { render } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "./rootReducer";

function renderWithProviders(ui, { reduxState } = {}) {
  const store = createStore(rootReducer, reduxState);
  return render(<Provider store={store}>{ui}</Provider>);
}
it("should take a snapshot", () => {
  const mockAppState = {
    imagesDetails: {
      rejectedIds: [],
      imageToAdd: {
        id: "",
        url: "",
      },
    },
  };
  const { asFragment } = renderWithProviders(<App />, mockAppState);

  expect(asFragment(<App />)).toMatchSnapshot();
});
