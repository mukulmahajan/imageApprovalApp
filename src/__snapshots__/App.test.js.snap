// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="sc-hKwDye jMXZju"
    >
      <div
        class="sc-eCImPb iyrDqV"
      >
        IMAGE APPROVAL APPLICATION
      </div>
      <p
        class="sc-pVTFL hFlOfJ"
      >
        APPROVED IMAGES (0)
      </p>
      <div
        class="sc-jrQzAO dYfsqQ"
      >
        <div
          class="sc-jRQBWg iFnklM"
        >
          <svg
            class="approved"
            data-testid="fetch-data"
            preserveAspectRatio="none"
            viewBox="0 0 42 42"
          >
            <path
              d="M37.059,16H26V4.941C26,2.224,23.718,0,21,0s-5,2.224-5,4.941V16H4.941C2.224,16,0,18.282,0,21s2.224,5,4.941,5H16v11.059 C16,39.776,18.282,42,21,42s5-2.224,5-4.941V26h11.059C39.776,26,42,23.718,42,21S39.776,16,37.059,16z"
            />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
          </svg>
        </div>
      </div>
      <div
        class="sc-gsDKAQ ljVwHa"
      >
        <svg
          class="addImage"
          data-testid="fetch-data"
          preserveAspectRatio="none"
          viewBox="0 0 42 42"
        >
          <path
            d="M37.059,16H26V4.941C26,2.224,23.718,0,21,0s-5,2.224-5,4.941V16H4.941C2.224,16,0,18.282,0,21s2.224,5,4.941,5H16v11.059 C16,39.776,18.282,42,21,42s5-2.224,5-4.941V26h11.059C39.776,26,42,23.718,42,21S39.776,16,37.059,16z"
          />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
        </svg>
      </div>
      <div
        class="sc-dkPtRN cjofvd"
      >
        <p>
          Click on the + in order to get image recommendations
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
